### ISSUE TITLE
Update MMG-Based-Transformer to populate two Event hubs - Metadata and Transformed Message

### ISSUE THREAD

**Issue by mscaldas2012** [CDCgov/data-exchange-hl7#319]
Transformation functions need to update two Event hubs upon success.
![image](https://user-images.githubusercontent.com/3239945/202267014-c7b021be-5da4-4324-9037-ba1d1ac082ed.png)

The payload for hl7-transformation-ok is similar to previous EHs:
``` json
{
   "content": "Base64(MSH|^~\&|...)",
   "meta_message_uuid": "",
   "summary": {
    	"current_status": "MESSAGE-TRANSFORMED-MMG-BASED"
    },
   "metadata": {
       "provenance": {
      	    "file_path": "abfss://container@storage/folder/filename.txt",
           "file_timestamp": "2022-10-12:58:00.000",
           "file_size": 1024,
           "file_uuid": "1234-56789",
           "single_or_batch": "SINGLE|BATCH",
           "message_index": 1,
           "message_hash": "asdf34-nweoru734",
           "ext_system_provider": "MVPS",
           "ext_original_file_name": "abc.txt",
           "ext_original_file_timestamp": ""
       },
       "processes": [
           {
            "meta_process_name": "Receiver",
            "meta_start_processing_time": "2022-10-01T13:00:00.000",
            "meta_end_processing_time": "2022-10-01T13:01:00.000",
            "meta_event_id": "1234-56789",
            "meta_event_timestamp": "2022-10-12:59:00.000",
            "status": "SUCCESS"
           },
 	    {
            "process_name": "Structure-Validator",
            "start_processing_time": "2022-10-01T13:00:00.000",
            "end_processing_time": "2022-10-01T13:01:00.000",
            "event_id": "1234-56789",
            "event_timestamp": "2022-10-12:59:00.000",
            "status": "SUCCESS",
	    "report": {
                 "error-count": 0,
                 "warning-count": 199,
                 "nist-report": { ... }
             }
           },
 	    {
            "process_name": "MMG-Validator",
            "start_processing_time": "2022-10-01T13:00:00.000",
            "end_processing_time": "2022-10-01T13:01:00.000",
            "event_id": "1234-56789",
            "event_timestamp": "2022-10-12:59:00.000",
            "status": "SUCCESS",
	    "report": {
                 "error-count": 0,
                 "warning-count": 199,
                 "mmg-report": { ... }
             }
           },
			 {
            "process_name": "MMG-PROCESSOR-MMG-BASED",
            "start_processing_time": "2022-10-01T13:00:00.000",
            "end_processing_time": "2022-10-01T13:01:00.000",
            "event_id": "1234-56789",
            "event_timestamp": "2022-10-12:59:00.000",
            "status": "SUCCESS"	    
        ]
    },
    "metadata_version":  "1"
}
```

The payload for the Transfomed message will contain only mesage_uuid and the transformed message:
``` josn
{

	"message_uuid": "1234-56768",
    "payload": {
		<<transformed message>>
    }

}


where <<transformed message>> is the JSON representation of a given schema.

for MMG-Based MMG, we will have the list of single Fields + all Repeating Elements.

Specific HL7 data types will have specific schemas.
For ST, NM, DT, TS, it's a simple value:

   "propertyA": "value"

For CE, CWE, will have specific fields for each element:     "propertyA:: {



   }
```

**Comment by rmharrison** [comment]
Decision: Not being implemented See @cosmycx ticket

**Comment by rmharrison** [comment]
@mscaldas2012 @cosmycx FYI / no action. Finally got around to writing this up.

### Decision
Option A. Which was implemented in #344

### Context
On 2022-11-29, the DEX team had implemented the "MMG Based Transformer" and "hl7-mmg-based-ok" EventHub topic. The team had NOT implemented error queues.

### Current state on 2022-11-29
![HL7 Issue 319 - current state.png](https://images.zenhubusercontent.com/62fe6e4e7c59c94524ab80ae/0b396f8f-5be3-41b3-891d-7e336a3e4ec8)

### Future State options for transformer error topics
We considered 4 options for representing the transformer error topics.
Everything upstream of the "MMG Based Transformer" is identical.
The area under consideration is highlighted with a green dashed-line box

#### Option A
![HL7 Issue 319 - Option A - Selected.png](https://images.zenhubusercontent.com/62fe6e4e7c59c94524ab80ae/705bffd3-b3fe-459c-9953-0c7888f9bc06)
The "same as usual" - One ok and err topic for each transformer function
Both metadata and data are contained in `ok`.
`err` queue does not contain data because the MMG transformation failed.

#### Option B
![HL7 Issue 319 - Option B.png](https://images.zenhubusercontent.com/62fe6e4e7c59c94524ab80ae/8c2faffe-21c3-4867-833a-62dd741c5a2e)
The "keep it all separate"
Metadata and data are kept seperate
ok metadata and err metadata are kept seperate

#### Option C
![HL7 Issue 319 - Option C.png](https://images.zenhubusercontent.com/62fe6e4e7c59c94524ab80ae/516750b9-ee28-4eef-8963-8e485d7cc574)
The "keep the transformer data all together"
Data for all transformers on the same `ok` topic
Error metadata for each transformer kept seperate

#### Option D
![HL7 Issue 319 - Option D.png](https://images.zenhubusercontent.com/62fe6e4e7c59c94524ab80ae/050efb08-129e-421c-b07b-15d9aa3d0ff9)
The "hybrid"
Keep the transformer data seperate
Keep the ok and err metadata on seperate topics. Each ok and err metadata topics has data for all transformers

### Discussion
We opted for Option A
Frankly, there isn't a good distinction between what is "data" vs what is "metadata."
If the "data" is only what was received from the sender, then the MMG-based model is metadata ==> The distinction between data and metadata for transformers doesn't make sense

If the "data" is the output of the transformer AND what we received from the client, what then is the "metadata" ==> The distinction between data and metadata for transformers doesn't make sense
